File: backend/NordicWebHub.Api/Models/SupportTicket.cs
using NordicWebHub.Api.Models.Enums;

namespace NordicWebHub.Api.Models;

public class SupportTicket
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public string CustomerId { get; set; } = string.Empty;

    public ApplicationUser Customer { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public SupportTicketStatus Status { get; set; } = SupportTicketStatus.Open;

    public TicketPriority Priority { get; set; } = TicketPriority.Normal;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? ClosedAt { get; set; }

    public ICollection<TicketReply> Replies { get; set; } = new List<TicketReply>();
}

File: frontend/nordicwebhub-client/src/types/supportTicket.ts
export type TicketStatus = 'Open' | 'InProgress' | 'WaitingForCustomer' | 'Closed'

export type TicketPriority = 'Low' | 'Medium' | 'High' | 'Urgent'

export type TicketReply = {
  id: number
  supportTicketId: number
  userId: string
  userEmail: string
  userFullName: string
  message: string
  createdAt: string
}

export type SupportTicket = {
  id: number
  companyId: number
  companyName: string
  customerId: string
  customerEmail: string
  title: string
  description: string
  status: TicketStatus
  priority: TicketPriority
  createdAt: string
  closedAt: string | null
  replies: TicketReply[]
}

export type CreateSupportTicketDto = {
  title: string
  description: string
  priority: TicketPriority
}

export type CreateTicketReplyDto = {
  message: string
}

export type UpdateTicketStatusDto = {
  status: TicketStatus
}

export type UpdateTicketPriorityDto = {
  priority: TicketPriority
}
